app/src/main/java/br/com/nespressohelper/daitan/nespressohelper/MainActivity.java

```diff
@@ -234,32 +170,36 @@ public void onClick(View view) {
 
     }
 
-    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    public byte[] setImageBitmapData(int index, int length, TypedArray drawables) {
+        if(index >= length) {
+            drawable = getDrawable(R.drawable.coffecapsule);
+            bitmap = ((BitmapDrawable) drawable).getBitmap();
+            stream = new ByteArrayOutputStream();
+            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+
+            return bitmapData = stream.toByteArray();
+        }else{
+            drawable = getDrawable(drawables.getResourceId(index, -1));//getDrawable(drawables.get(i));
+            bitmap = ((BitmapDrawable) drawable).getBitmap();
+            stream = new ByteArrayOutputStream();
+            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+
+            return bitmapData = stream.toByteArray();
```

Comment:
This seems to be the same code we have in AddProductActivity. You could maybe extract it to a BitmapUtils class or something similar.